### PR TITLE
Add rasam

### PR DESCRIPTION
**Pull request by JoelJaeschke**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by JoelJaeschke** [comment]
@conda-forge/help-c-cpp, ready for review!

File: recipes/rasam/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+build:
+  number: 0
+
+requirements:
+  build:
+    - make
+    - cmake
+    - {{ compiler('cxx') }}
+    - libgdal >=3.7.0
```
Comment (beenje) [inline recipes/rasam/meta.yaml:20]: `libgdal` is only needed under `host`.
It shouldn't be under `build`.

And you don't have to set it under `run` as the gdal recipe is defining [run_exports](https://github.com/conda-forge/gdal-feedstock/blob/e80378815e7683f0288880f5862bb689a961a9d8/recipe/meta.yaml#L75): having it in host will automatically add it as run dependency

File: recipes/rasam/build.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
+set -e
+
+BUILD_DIR="build"
+
+mkdir -p ${BUILD_DIR}
+cd ${BUILD_DIR}
+
+# Create build files
+cmake .. -DCMAKE_INSTALL_PREFIX=$PREFIX -DCMAKE_BUILD_TYPE=Release
```
Comment (beenje) [inline recipes/rasam/build.sh:10]: I'd recommend to also pass `${CMAKE_ARGS}` to define some conda-forge specific variables.

**Review by beenje — CHANGES_REQUESTED** [review]

File: recipes/rasam/meta.yaml (hunk 1 above)
Comment (JoelJaeschke) [reply recipes/rasam/meta.yaml]: Sorry for the late reply, I knew I missed something. Implemented your changes now

File: recipes/rasam/build.sh (hunk 2 above)
Comment (JoelJaeschke) [reply recipes/rasam/build.sh]: Done

**Pull request merged**